quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/logging/ClearMappedDiagnosticContextFilter.java

```diff
@@ -0,0 +1,26 @@
+package org.keycloak.quarkus.runtime.logging;
+
+import jakarta.annotation.Priority;
+import jakarta.ws.rs.container.ContainerRequestContext;
+import jakarta.ws.rs.container.ContainerResponseContext;
+import jakarta.ws.rs.container.ContainerResponseFilter;
+import jakarta.ws.rs.ext.Provider;
+import org.keycloak.logging.MappedDiagnosticContextUtil;
+
+import java.io.IOException;
+
+/**
+ * Response filter that clears custom properties from MDC.
+ *
+ * @author <a href="mailto:b.eicki@gmx.net">Björn Eickvonder</a>
+ */
+@Provider
+@Priority(0)
+public class ClearMappedDiagnosticContextFilter implements ContainerResponseFilter {
```

Comment:
I think this class can be removed if you implement `DefaultMappedDiagnosticContextProvider.close()` to remove the keys from the `MDC`. When the session is closed, the MDC will clear automatically. 
The changes in `ScheduledTaskRunner` can be reverted too, if `close()` removes the keys.